hackerRankProblems/pair_of_socks.js

```diff
@@ -0,0 +1,35 @@
+// John works at a clothing store. He has a large pile of socks that he must pair by color for sale. Given an array of integers representing the color of each sock, determine how many pairs of socks with matching colors there are.
+// For example, there are n = 7 socks with colors arr = [1,2,1,2,1,3,2]. There is one pair of color  and one of color . There are three odd socks left, one of each color. The number of pairs is .
+// Function Description
+// Complete the sockMerchant function in the editor below. It must return an integer representing the number of matching pairs of socks that are available.
+// sockMerchant has the following parameter(s):
+// n: the number of socks in the pile
+// ar: the colors of each sock
+
+function pairOfSocks(n, arr) {
+  const count = {};
+  let pairs = 0;
+
+  arr.forEach(element => {
+    if (count[element]) {
+      count[element] -= 1;
```

Comment:
sweet trick!

only 2 comments. 
1) `count` may not be the best name here, maybe something like `singleMap`
2) on line 14 you're leveraging the falseness of of `undefined` and `0` in javascript. be sure to talk about this slick move. in most places, you'll be using `TypeScript` or `Flow` or something that will likely frown on this slickness. It may be even considered dangerous. It may be less dangerous and more readable if you explicitly check for `undefined` and `0`. Either which way you do. Just acknowledge the alternative and trade-offs. Example: if you were to explicitly check for `undefined` you can say "I *could* leverage the falseness of undefined and 0 in JavaScript, but I find it more readable and less dangerous to be explicit"

Reply:
#1 noted.
#2 I should definitely be more explicit now on. I always knew this, but guess needed a reminder. Thanks!